### DISCUSSION TITLE
Small errors in the book

### DISCUSSION THREAD

**Discussion by ArnaudLier** [symfony/symfony#46435] [Symfony Book: The Fast Track] [answered]
Just reporting some errors that I encountered while following the book:

https://symfony.com/doc/current/the-fast-track/en/9-backend.html#installing-more-dependencies
You don't build the EasyAdmin assets with `npm run build`

https://symfony.com/doc/current/the-fast-track/en/17-tests.html#configuring-the-test-environment
You set the `AKISMET_KEY` for the `test` environment but you don't need it?

https://symfony.com/doc/current/the-fast-track/en/17-tests.html#working-with-a-test-database:
You set the environment to `prod` but it should be `test`

**Comment by srbetasign** [comment]
I may add another mistake seen in the documentation of SF 6.0 concerning the configuration of Validation : https://symfony.com/doc/6.0/reference/configuration/framework.html#validation

Aa seen in the screenshot below,  the default value of validation.mapping.paths is ['config/validation/']
![image](https://user-images.githubusercontent.com/10088615/180462763-3cf9824d-c3f3-4c5c-99ae-4f10e7bd7250.png)

But the real default value is ['config/validator/'].

**Reply by fabpot** [reply]
> I may add another mistake seen in the documentation of SF 6.0 concerning the configuration of Validation : https://symfony.com/doc/6.0/reference/configuration/framework.html#validation
> 
> But the real default value is ['config/validator/'].

@srbetasign  You should report this here: https://github.com/symfony/symfony-docs.

**Comment by lyrixx** [comment]
cc @fabpot

**Comment by bruteforks** [comment]
another small thing from book I've just spent a long time on -- needed to `sudo apt-get install php-pgsql` for the command  `symfony console make:entity Conference` otherwise displayed driver errors

**Reply by fabpot** [reply]
> another small thing from book I've just spent a long time on -- needed to `sudo apt-get install php-pgsql` for the command `symfony console make:entity Conference` otherwise displayed driver errors

This dependency is listed as the ones you need to install in chapter 1: https://symfony.com/doc/current/the-fast-track/en/1-tools.html#php

**Reply by bruteforks** [reply]
> This dependency is listed as the ones you need to install in chapter 1: https://symfony.com/doc/current/the-fast-track/en/1-tools.html#php

ah, i'm so sorry. don't know how i missed this

**Comment by fabpot** [comment, marked as answer]
> https://symfony.com/doc/current/the-fast-track/en/17-tests.html#working-with-a-test-database: You set the environment to `prod` but it should be `test`

Will be fixed in the 6.2 book.

**Reply by MaximumRGB** [reply]
https://symfony.com/doc/current/the-fast-track/en/16-spam.html
The API call response (endpoint _https://{AKISMET_KEY}.rest.akismet.com/1.1/comment-check_) is **true** if it finds a comment to be spam. However, there is no _x-akismet-pro-tip header_. So, the **getSpamScore()** method of **SpamChecker** class does not return 2, even if comment_author='_akismet-guaranteed-spam_' or comment_author_email='_akismet-guaranteed-spam@example.com_'.

**Comment by wail-hammani** [comment]
Hello everyone, i'am new here and in symfony and i'm following the fast track book. While trying to deploy to platform.sh by running command : 
`symfony cloud:deploy` 
i see in the terminal that everything is ok but on the platform dashboard i see that the deployment failed and returns this error :

` W: unable to run /app/.global/bin/composer2 install --prefer-dist --optimize-autoloader --apcu-autoloader --classmap-authoritative --no-progress --no-ansi --no-interaction
  
  E: Error building project: Step failed with status code 1.

E: Error: Unable to build application, aborting.`

**Reply by fabpot** [reply]
Can you share the full output?

**Reply by wail-hammani** [reply]
@fabpot  I found the error and it was because of the database URL env variable which was empty. Thanks for your reply

**Comment by jibla** [comment]
https://symfony.com/doc/6.2/the-fast-track/en/13-lifecycle.html

Here it is said  - `When that happens, the setCreatedAtValue() method is called and the current date and time is used for the value of the createdAt property.`, however actually in the code snippet shown, the value is directly assigned to createdAt value. I believe the code snippet should be fixed as the right way is to call the setCreatedAtValue() method.

![image](https://user-images.githubusercontent.com/475343/230721714-42e326f5-9b10-4b6d-a240-62802f91df75.png)

**Reply by stof** [reply]
Well, the second part (after the `and`) describes what happens when this method is called.

**Reply by jibla** [reply]
Sure, you are right, the code and the text are in sync. My bad.
The text mentions the `setCreatedAtValue()` itself, which, somehow I just mixed with the entity's `setCreatedAt(\DateTimeImmutable $createdAt): self` method.

However, still, perhaps its a better way to use `setCreatedAt()` internal setter to set the current DateTime, instead of directly assigning the value? Like this:
```php
#[ORM\PrePersist]
public function setCreatedAtValue()
{
    $this->setCreatedAt(new \DateTimeImmutable());
}
```

**Reply by stof** [reply]
there is no benefit to that. Accessing the private property directly is perfectly fine (thus, there is no reason to even keep a `setCreatedAt` method in the entity IMO)

**Reply by jibla** [reply]
I agree, in this case, there is no difference. However, in general, it's better practice to encapsulate property assigning in a method, for cases when you need to manipulate value before assigning every time.
